.github/workflows/main.yml

```diff
@@ -16,6 +16,26 @@ on:
 
 jobs:
 
+  mingw:
+    runs-on: windows-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        # Fetch all history to make sure tags are
+        # included (used for version creation)
+        fetch-depth: 0
+    - uses: seanmiddleditch/gha-setup-ninja@v3
```

Comment:
Is there a canonical source for Ninja? I worry that using somebody's fork is fragile.

Reply:
Most github actions for setting these things up are created by users like this. I've used this particular action for setting up ninja a number of times and it's pretty widely depended on. I see the concern but I'd argue that this is as stable as it can get for github actions